meson_config.h.in

```diff
@@ -0,0 +1,179 @@
+#undef HAVE_STDLIB_H
+
+/* Define to 1 if you have the `strchrnul' function. */
+#undef HAVE_STRCHRNUL
+
+/* Define to 1 if you have the `strdup' function. */
+#undef HAVE_STRDUP
+
+/* Define to 1 if you have the `strerror' function. */
+#undef HAVE_STRERROR
+
+/* Define to 1 if you have the <strings.h> header file. */
+#undef HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#undef HAVE_STRING_H
+
+/* Define to 1 if you have the `strndup' function. */
+#undef HAVE_STRNDUP
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#undef HAVE_SYS_STAT_H
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#undef HAVE_SYS_TYPES_H
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#undef HAVE_UNISTD_H
+
+/* Define to 1 if you have the `vprintf' function. */
+#undef HAVE_VPRINTF
+
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* If using the C implementation of alloca, define if you know the
+   direction of stack growth for your system; otherwise it will be
+   automatically deduced at run-time.
+	STACK_DIRECTION > 0 => grows toward higher addresses
+	STACK_DIRECTION < 0 => grows toward lower addresses
+	STACK_DIRECTION = 0 => direction of growth unknown */
+#undef STACK_DIRECTION
+
+/* Define to 1 if you have the ANSI C header files. */
+#undef STDC_HEADERS
+
+/* Version number of package */
+#undef VERSION
+
+/* Enable GNU extensions on systems that have them.  */
+#ifndef _GNU_SOURCE
+# undef _GNU_SOURCE
+#endif
+
+/* Define to empty if `const' does not conform to ANSI C. */
+#undef const
+
+/* Define to `__inline__' or `__inline' if that's what the C compiler
+   calls it, or to nothing if 'inline' is not supported under any name.  */
+#ifndef __cplusplus
+#undef inline
+#endif
+
+/* Define to `unsigned' if <sys/types.h> does not define. */
+#undef size_t
+
+#mesondefine HAVE_CONFIG_H
+#mesondefine HAVE_UNISTD_H
+#mesondefine HAVE_ALLOCA_H
+#mesondefine HAVE_MALLOC_H
+#mesondefine HAVE_LIBINTL_H
+
+#mesondefine HAVE_ASPRINTF
+#mesondefine HAVE_STRCHRNUL
+#mesondefine HAVE_STRNDUP
+
+#mesondefine HAVE_DECL_PROGRAM_INVOCATION_NAME
+#mesondefine HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME
+
+#mesondefine HAVE_DECL_FWRITE_UNLOCKED
+#mesondefine HAVE_DECL_CLEARERR_UNLOCKED
+#mesondefine HAVE_DECL_FEOF_UNLOCKED
+#mesondefine HAVE_DECL_FERROR_UNLOCKED
+#mesondefine HAVE_DECL_FFLUSH_UNLOCKED
+#mesondefine HAVE_DECL_FGETS_UNLOCKED
+#mesondefine HAVE_DECL_FPUTC_UNLOCKED
+#mesondefine HAVE_DECL_FPUTS_UNLOCKED
+#mesondefine HAVE_DECL_FLOCKFILE
+#mesondefine HAVE_DECL_PUTC_UNLOCKED
```

Comment:
You aren't testing for all of these in meson.build. And as mentioned, most usage should simply be switched to the locking version, in cases where windows doesn't provide the function.